pkg/controller/installation/core_controller.go

```diff
@@ -108,32 +108,43 @@ func newReconciler(mgr manager.Manager, opts options.AddOptions) (*ReconcileInst
 
 	statusManager := status.New(mgr.GetClient(), "calico", opts.KubernetesVersion)
 
-	// The typhaAutoscaler needs a clientset.
+	// The typhaAutoscaler and calicoWindowsUpgrader need a clientset.
 	cs, err := kubernetes.NewForConfig(mgr.GetConfig())
 	if err != nil {
 		return nil, err
 	}
 
-	// Create a Typha autoscaler.
 	nodeListWatch := cache.NewListWatchFromClient(cs.CoreV1().RESTClient(), "nodes", "", fields.Everything())
+
+	// Create a Typha autoscaler.
 	typhaListWatch := cache.NewListWatchFromClient(cs.AppsV1().RESTClient(), "deployments", "calico-system", fields.OneTermEqualSelector("metadata.name", "calico-typha"))
 	typhaScaler := newTyphaAutoscaler(cs, nodeListWatch, typhaListWatch, statusManager)
 
+	requestChan := make(chan utils.ReconcileRequest)
+
+	// Create a Calico Windows upgrader.
+	calicoWindowsUpgrader := newCalicoWindowsUpgrader(cs, mgr.GetClient(), nodeListWatch, statusManager, requestChan)
```

Comment:
I still think we should create the indexer outside of `newTyphaAutoscaler` and `newCalicoWindowsUpgrader`.
I think we could run a rather tight loop (like 1/sec) for windows and if any of the values 	w.nodesToUpgrade, w.nodesUpgrading, or w.nodesFinishedUpgrade changed then trigger a reconcile.
WDYT?

Reply:
I started down this track with 13ab000ad193161caba41aee644be621681f5c80 ... maybe we can chat about this tomorrow a bit more.